src/action/list_directory.rs

```diff
@@ -0,0 +1,533 @@
+            st_dev: umetadata.dev() as u32,
+            st_nlink: umetadata.nlink() as u32,
+            st_uid: umetadata.uid() as u32,
+            st_gid: umetadata.gid() as u32,
+            st_size: umetadata.size(),
+            st_atime: umetadata.atime() as u64,
+            st_mtime: umetadata.mtime() as u64,
+            st_ctime: umetadata.ctime() as u64,
+            st_blocks: umetadata.blocks() as u32,
+            st_blksize: umetadata.blksize() as u32,
+            st_rdev: umetadata.rdev() as u32,
+            st_flags_linux:
+            get_linux_flags(file_path).unwrap_or_default() as u32,
+            symlink: match umetadata.file_type().is_symlink() {
+                true => match fs::read_link(file_path) {
+                    Ok(file) => Some(file.to_string_lossy().to_string()),
+                    _ => None,
+                }
+                false => None
+            },
+            pathspec: PathSpec {
+                path_options: Some(PathOption::CaseLiteral),
+                pathtype: PathType::OS,
+                path: file_path.clone(),
+            },
+        })?;
+    }
+    Ok(())
+}
+
+/// Converts integer from proto to human-readable enum type
+fn get_enum_path_options(option: &Option<i32>) -> Option<PathOption> {
+    match option {
+        Some(poption) => match poption {
+            0 => Some(PathOption::CaseInsensitive),
+            1 => Some(PathOption::CaseLiteral),
+            2 => Some(PathOption::Recursive),
+            3 => Some(PathOption::Regex),
+            _ => None
+        },
+        _ => None,
+    }
+}
+
+/// Converts integer from proto to human-readable enum type
+fn get_enum_path_type(option: &Option<i32>) -> Option<PathType> {
+    match option {
+        Some(ptype) => match ptype {
+            0 => Some(PathType::OS),
+            1 => Some(PathType::TSK),
+            2 => Some(PathType::Registry),
+            3 => Some(PathType::TMPFile),
+            4 => Some(PathType::NTFS),
+            _ => None,
+        },
+        _ => None,
+    }
+}
+
+fn get_path(path: &Option<String>) -> PathBuf {
+    match path {
+        Some(string_path) => {
+            if string_path.is_empty() {
+                PathBuf::from("/")
+            } else {
+                PathBuf::from(string_path)
+            }
+        }
+        _ => PathBuf::from("/"),
+    }
+}
+
+/// Fills st_linux_flags field
+fn get_linux_flags(path: &PathBuf) -> Option<c_long> {
+    let file = match File::open(path) {
+        Ok(file) => file,
+        Err(_) => return None,
+    };
+    let mut linux_flags: c_long = 0;
+    let linux_flags_ptr: *mut c_long = &mut linux_flags;
+    unsafe {
+        match ioctls::fs_ioc_getflags(file.as_raw_fd(), linux_flags_ptr) {
+            0 => Some(linux_flags),
+            _ => None,
+        };
+    }
+    Some(linux_flags)
+}
+
+/// Converts enum type back to integer to pass to the proto
+fn get_int_path_options(pathspec: &PathSpec) -> Option<i32> {
```

Comment:
Is there a reason for this not being a method of `PathSpec`?

Reply:
I removed all such functions.